frappe/public/js/frappe/ui/colors.js

```diff
@@ -39,7 +39,8 @@ frappe.ui.color = {
 		};
 
 		if(Object.keys(shades).includes(shade)) {
-			return frappe.ui.color_map[color_name][shades[shade]];
+			const color = this.get_color(color_name);
+			return color[shades[shade]];
```

Comment:
```suggestion
			const color = this.get_color(color_name);
			return color ? color[shades[shade]] : color_name;
```

Reply:
I think optional chaining could be more readable here:

```suggestion
			const color = this.get_color(color_name);
			return color?.shades?.shade || color_name;
```